lib/reader.js

```diff
@@ -68,20 +66,39 @@ module.exports = class Reader extends pack {
         });
 
         // Set up the pipeline
-        const order = new Order();
-        const sort = depsSort({ dedupe: true });
-
-        merged.on('error', (error) => {
-            this.emit('error', error);
-        })
-              .pipe(order)
-              .on('error', (error) => {
-                  this.emit('error', error);
-              })
-              .pipe(sort)
-              .on('error', (error) => {
-                  this.emit('error', error);
-              })
-              .pipe(this);
+        merged
+            .on('error', (error) => {
+                this.emit('error', error);
+            })
+            .pipe(sortAndDedupe())
+            .on('error', (error) => {
+                this.emit('error', error);
+            })
+            .pipe(this);
     }
 };
+
+
+function sortAndDedupe () {
+    const rows = new Map();
+    return through.obj(write, end);
+
+    function write (row, enc, next) {
+        if (!rows.has(row.id)) {
+            rows.set(row.id, row);
+        }
+        next();
+    }
+
+    function end () {
+        Array
+            .from(rows.values())
+            .sort(compareId)
+            .forEach(row => this.push(row));
+        this.push(null);
+    }
+}
+
+function compareId (a, b) {
+    return a.id < b.id ? -1 : 1;
```

Comment:
The ids can be the same because of merging of feed-streams from different packages. e.g. bluebird will have the same id.